tests/scripts/deploy-validate-vault.sh

```diff
@@ -276,6 +276,34 @@ function validate_osd_secret {
   fi
 }
 
+function validate_key_rotation() {
+  local backend_path=$1
+  pvc_name=$(kubectl get pvc -n rook-ceph -l ceph.rook.io/setIndex=0 -o jsonpath='{.items[0].metadata.name}')
+  key_name="rook-ceph-osd-encryption-key-$pvc_name"
+  cmd="vault kv get -format=json $backend_path/$key_name"
+  old_key=$(kubectl exec vault-0 -- sh -c "$cmd" | jq -r ".data.\"$key_name\"")
+
+  runtime="3 minutes"
+  endtime=$(date -ud "$runtime" +%s)
+  while [[ $(date -u +%s) -le $endtime ]]; do
+    echo "Time Now: $(date +%H:%M:%S)"
+    new_key=$(kubectl exec vault-0 -- sh -c "$cmd" | jq -r ".data.\"$key_name\"")
+    if [ "$old_key" != "$new_key" ]; then
+      echo "encryption passphrase is successfully rotated"
+      exit 0
+    fi
+    echo "encryption passphrase is not rotated, sleeping for 10 seconds"
+    sleep 10s
+  done
+  new_key=$(kubectl exec vault-0 -- sh -c "$cmd" | jq -r ".data.\"$key_name\"")
+  if [ "$old_key" == "$new_key" ]; then
+    echo "encryption passphrase is not rotated"
+    exit 1
+  else
+    echo "encryption passphrase is successfully rotated"
+  fi
+}
```

Comment:
```suggestion
runtime=180  # 3 minutes in seconds
endtime=$((SECONDS + runtime))
 while [ $SECONDS -le $endtime ]; do
    echo "Time Now: $(date +%H:%M:%S)"
    local new_key=$(kubectl exec vault-0 -- sh -c "$cmd" | jq -r ".data.\"$key_name\"")
    
    if [ "$old_key" != "$new_key" ]; then
      echo "Encryption passphrase successfully rotated."
      exit 0
    fi
    
    echo "Encryption passphrase not rotated, sleeping for 10 seconds."
    sleep 10
  done
  
  echo "Encryption passphrase not rotated."
  exit 1
}
```

can we do like as suggested as we are using seconds to sleep and also this reduces a few complex checks?